### DISCUSSION TITLE
[BUG?] `isModuleDeclaration` has been deprecated, please migrate to `isImportOrExportDeclaration`

### DISCUSSION THREAD

**Discussion by nanakochi123456** [mastodon/mastodon#25787] [General]
$  RAILS_ENV=production bundle exec rails db:migrate
$  RAILS_ENV=production bundle exec rake tmp:cache:clear
$  RAILS_ENV=production bundle exec rails assets:precompile
yarn install v1.22.19
[1/6] Validating package.json...
[2/6] Resolving packages...
[3/6] Fetching packages...
warning Pattern ["strip-ansi@^6.0.1"] is trying to unpack in the same destination "/usr/local/www/mastodon/.cache/yarn/v6/npm-strip-ansi-cjs-6.0.1-9e26c63d30f53443e9489495b2105d37b67a85d9-integrity/node_modules/strip-ansi-cjs" as pattern ["strip-ansi-cjs@npm:strip-ansi@^6.0.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["string-width@^4.1.0"] is trying to unpack in the same destination "/usr/local/www/mastodon/.cache/yarn/v6/npm-string-width-cjs-4.2.3-269c7117d27b05ad2e536830a8ec895ef9c6d010-integrity/node_modules/string-width-cjs" as pattern ["string-width-cjs@npm:string-width@^4.2.0"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["strip-ansi@^6.0.0"] is trying to unpack in the same destination "/usr/local/www/mastodon/.cache/yarn/v6/npm-strip-ansi-cjs-6.0.1-9e26c63d30f53443e9489495b2105d37b67a85d9-integrity/node_modules/strip-ansi-cjs" as pattern ["strip-ansi-cjs@npm:strip-ansi@^6.0.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["string-width@^4.2.3"] is trying to unpack in the same destination "/usr/local/www/mastodon/.cache/yarn/v6/npm-string-width-cjs-4.2.3-269c7117d27b05ad2e536830a8ec895ef9c6d010-integrity/node_modules/string-width-cjs" as pattern ["string-width-cjs@npm:string-width@^4.2.0"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["string-width@^1.0.2 || 2 || 3 || 4"] is trying to unpack in the same destination "/usr/local/www/mastodon/.cache/yarn/v6/npm-string-width-cjs-4.2.3-269c7117d27b05ad2e536830a8ec895ef9c6d010-integrity/node_modules/string-width-cjs" as pattern ["string-width-cjs@npm:string-width@^4.2.0"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["string-width@^4.2.0"] is trying to unpack in the same destination "/usr/local/www/mastodon/.cache/yarn/v6/npm-string-width-cjs-4.2.3-269c7117d27b05ad2e536830a8ec895ef9c6d010-integrity/node_modules/string-width-cjs" as pattern ["string-width-cjs@npm:string-width@^4.2.0"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["wrap-ansi@^7.0.0"] is trying to unpack in the same destination "/usr/local/www/mastodon/.cache/yarn/v6/npm-wrap-ansi-cjs-7.0.0-67e145cff510a6a6984bdf1152911d69d2eb9e43-integrity/node_modules/wrap-ansi-cjs" as pattern ["wrap-ansi-cjs@npm:wrap-ansi@^7.0.0"]. This could result in non-deterministic behavior, skipping.
[4/6] Linking dependencies...
warning " > emoji-mart@3.0.1-j" has incorrect peer dependency "react@^0.14.0 || ^15.0.0-0 || ^16.0.0 || ^17.0.0".
warning " > react-motion@0.5.2" has incorrect peer dependency "react@^0.14.9 || ^15.3.0 || ^16.0.0".
warning " > react-notification@6.8.5" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning " > react-router-scroll-4@1.0.0-beta.2" has incorrect peer dependency "react@^15.0.0 || ^16.0.0".
warning " > react-router-scroll-4@1.0.0-beta.2" has incorrect peer dependency "react-dom@^15.0.0 || ^16.0.0".
warning " > react-swipeable-views@0.14.0" has incorrect peer dependency "react@^15.3.0 || ^16.0.0 || ^17.0.0".
warning "react-swipeable-views > react-swipeable-views-utils > react-event-listener@0.6.6" has incorrect peer dependency "react@^16.3.0".
warning Workspaces can only be enabled in private projects.
[5/6] Building fresh packages...
warning Error running install script for optional dependency: "/usr/local/www/mastodon/node_modules/bufferutil: Command failed.
Exit code: 1
Command: node-gyp-build
Arguments:
Directory: /usr/local/www/mastodon/node_modules/bufferutil
Output:
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn node-gyp ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn node-gyp',
  path: 'node-gyp',
  spawnargs: [ 'rebuild' ]
}

Node.js v18.16.0"
info This module is OPTIONAL, you can safely ignore this error
warning Error running install script for optional dependency: "/usr/local/www/mastodon/node_modules/utf-8-validate: Command failed.
Exit code: 1
Command: node-gyp-build
Arguments:
Directory: /usr/local/www/mastodon/node_modules/utf-8-validate
Output:
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn node-gyp ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn node-gyp',
  path: 'node-gyp',
  spawnargs: [ 'rebuild' ]
}

Node.js v18.16.0"
info This module is OPTIONAL, you can safely ignore this error
[6/6] Cleaning modules...
Done in 9.41s.
Compiling...
Compilation failed:
`isModuleDeclaration` has been deprecated, please migrate to `isImportOrExportDeclaration`
    at isModuleDeclaration (/usr/local/www/mastodon/node_modules/@babel/types/lib/validators/generated/index.js:2740:35)
    at PluginPass.Program (/usr/local/www/mastodon/node_modules/babel-plugin-lodash/lib/index.js:102:44)
ModuleNotFoundError: Module not found: Error: Can't resolve 'intl' in '/usr/local/www/mastodon/app/javascript/mastodon'
    at /usr/local/www/mastodon/node_modules/webpack/lib/Compilation.js:925:10
    at /usr/local/www/mastodon/node_modules/webpack/lib/NormalModuleFactory.js:401:22
    at /usr/local/www/mastodon/node_modules/webpack/lib/NormalModuleFactory.js:130:21
    at /usr/local/www/mastodon/node_modules/webpack/lib/NormalModuleFactory.js:224:22
    at /usr/local/www/mastodon/node_modules/neo-async/async.js:2830:7
    at /usr/local/www/mastodon/node_modules/neo-async/async.js:6877:13
    at /usr/local/www/mastodon/node_modules/webpack/lib/NormalModuleFactory.js:214:25
    at /usr/local/www/mastodon/node_modules/enhanced-resolve/lib/Resolver.js:213:14
    at /usr/local/www/mastodon/node_modules/enhanced-resolve/lib/Resolver.js:285:5
    at eval (eval at create (/usr/local/www/mastodon/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at /usr/local/www/mastodon/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7

etc..

by FreeBSD Package is newer.

**Comment by nanakochi123456** [comment]
ver=4,1,3

**Comment by renchap** [comment]
`isModuleDeclaration has been deprecated, please migrate to isImportOrExportDeclaration` is just a warning, you can ignore it.

Your issue comes from your installed Node version. Mastodon does not support Node 18 yet, you need to use Node 16.

**Reply by nanakochi123456** [reply]
Below this log is written as follows, unable to precompile, We're sorry, but something went wrong on our end.

              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.js doesn't exist
            .jsx
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.jsx doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.tsx doesn't exist
            .sass
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.sass doesn't exist
            .scss
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.scss doesn't exist
            .css
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.css doesn't exist
            .module.sass
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.module.sass doesn't exist
            .module.scss
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.module.scss doesn't exist
            .module.css
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.module.css doesn't exist
            .png
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.png doesn't exist
            .svg
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.svg doesn't exist
            .gif
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.gif doesn't exist
            .jpeg
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.jpeg doesn't exist
            .jpg
              Field 'browser' doesn't contain a valid alias configuration
              /usr/local/www/mastodon/node_modules/intl.jpg doesn't exist
            as directory

**Reply by nanakochi123456** [reply]
and now using node18 packages.

**Comment by nanakochi123456** [comment]
Using node16

We're sorry, but something went wrong on our end

(nginx log)
2023/07/07 19:50:38 [error] 3457#100808: *1 kevent() reported that connect() failed (61: Connection refused) while connecting to upstream, client: 192.168.0.253, server: host, request: "GET /api/v1/announcements HTTP/1.1", upstream: "http://127.0.0.1:3000/api/v1/announcements", host: "host", referrer: "https://host/home"

**Reply by nanakochi123456** [reply]
There is no toot in the browser and it is black. The icon is completely black as well.
However, only the title bar is displayed.
